uint8 PROBLEM_GENERATION = 0
uint8 WAITING_PROBLEM = 1
uint8 PLAN_GENERATION = 2
uint8 WAITING_PLAN = 3
uint8 PLAN_PARSING = 4
uint8 WAITING_PARSING = 5
uint8 DISPATCH_ACTION = 6
uint8 WAITING_REACHING_GOAL = 7
uint8 ERROR = 99
